internal/pkg/agent/application/managed_mode.go

```diff
@@ -168,13 +168,21 @@ func (m *managedConfigManager) Run(ctx context.Context) error {
 		}
 	}
 
+	var stateFetcher fleetgateway.StateFetcher
+	if m.cfg.Fleet.Checkin.IsModeOnStateChanged() {
```

Comment:
Let's log which fetcher we've decided to use in here.

Reply:
Added the checkin mode in log.